src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -0,0 +1,250 @@
+          Thread thread = defaultFactory.newThread(r);
+          thread.setName("gpu-timezone-database-0");
+          thread.setDaemon(true);
+          return thread;
+        }
+      });
+    instance.loadData(executor);
+  }
+
+  public static void shutdown() {
+    instance.getHostFixedTransitions().close();
+  }
+
+  public static ColumnVector fromTimestampToUtcTimestamp(ColumnVector input, ZoneId currentTimeZone) {
+    // TODO: Remove this check when all timezones are supported
+    // (See https://github.com/NVIDIA/spark-rapids/issues/6840)
+    if (!isSupportedTimeZone(currentTimeZone)) {
+      throw new IllegalArgumentException(String.format("Unsupported timezone: %s",
+          currentTimeZone.toString()));
+    }
+    Integer tzIndex = instance.getZoneIDMap().get(currentTimeZone.normalized().toString());
+    Table transitions = instance.getTransitions();
+    ColumnVector result = new ColumnVector(convertTimestampColumnToUTC(input.getNativeView(),
+        transitions.getNativeView(), tzIndex));
+    transitions.close();
+    return result;
+  }
+  
+  public static ColumnVector fromUtcTimestampToTimestamp(ColumnVector input, ZoneId desiredTimeZone) {
+    // TODO: Remove this check when all timezones are supported
+    // (See https://github.com/NVIDIA/spark-rapids/issues/6840)
+    if (!isSupportedTimeZone(desiredTimeZone)) {
+      throw new IllegalArgumentException(String.format("Unsupported timezone: %s",
+          desiredTimeZone.toString()));
+    }
+    Integer tzIndex = instance.getZoneIDMap().get(desiredTimeZone.normalized().toString());
+    Table transitions = instance.getTransitions();
+    ColumnVector result = new ColumnVector(convertUTCTimestampColumnToTimeZone(input.getNativeView(),
+        transitions.getNativeView(), tzIndex));
+    transitions.close();
+    return result;
+  }
+  
+  // TODO: Deprecate this API when we support all timezones 
+  // (See https://github.com/NVIDIA/spark-rapids/issues/6840)
+  public static boolean isSupportedTimeZone(ZoneId desiredTimeZone) {
+    String id = desiredTimeZone.normalized().getId();
+    return instance.getZoneIDMap().containsKey(id);
+  }
+  
+  private void loadData(Executor executor) throws IllegalStateException {
+    // Start loading the data in separate thread and return
+    try {
+      executor.execute(this::doLoadData);
+    } catch (RejectedExecutionException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private void doLoadData() {
+    synchronized (this) {
+      try {
+        Map<String, Integer> zoneIdToTable = new HashMap<>();
+        List<List<HostColumnVector.StructData>> masterTransitions = new ArrayList<>();
+        for (String tzId : TimeZone.getAvailableIDs()) {
+          ZoneId zoneId;
+          try {
+            zoneId = ZoneId.of(tzId).normalized(); // we use the normalized form to dedupe
+          } catch (ZoneRulesException e) {
+            continue;
```

Comment:
It actually would never happen in this case.  This try/catch might have been added by the IDE, but it's not necessary here. This is an exception that occurs when you pass in an invalid Timezone Id to this method that can't be found in the IANA database.

Reply:
Should add, that the source of this is Java itself (`TimeZone.getAvailableIds()`), in which the data is coming from the same place.

Reply:
Never mind, it seems that this data is somewhat inconsistent (probably because there is some ambiguity to be resolved, ie the 3-letter abbreviations which are available but deprecated). Maybe we should file a followup issue to handle that case?

Reply:
This all depends on how many time zones we are going to be able to support, and if we end up supporting them dynamically or not. A comment and a follow on issue should be fine. I am happy to have them skipped for now, but eventually we will need a full fix.